analysis/src/python/data_analysis/preprocessing/preprocess_hidden_templates.py

```diff
@@ -0,0 +1,120 @@
+import argparse
+from collections import defaultdict
+from enum import Enum
+from pathlib import Path
+from typing import Dict, Optional
+
+import pandas as pd
+
+from analysis.src.python.data_analysis.model.column_name import SubmissionColumns
+from analysis.src.python.data_analysis.utils.code_utils import merge_lines_to_code, split_code_to_lines
+from analysis.src.python.utils.df_utils import apply, read_df, write_df
+
+
+class TemplateBlock(Enum):
+    HEADER = '::header'
+    CODE = '::code'
+    FOOTER = '::footer'
+
+    @staticmethod
+    def from_value(value: str) -> Optional['TemplateBlock']:
+        try:
+            return TemplateBlock(value)
+        except ValueError:
+            return None
+
+
+class Template:
```

Comment:
Please, add several tests, because it potentially can have bugs

Reply:
Could you please also extend the template in the template column? I mean it would be great if we have a column with the full template without additional user code